taskEight/src/by/dorosinec/traning/product/Product.java

```diff
@@ -0,0 +1,42 @@
+package by.dorosinec.traning.product;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * Structure of products. This structure describes all fields of products
+ */
+public class Product {
+    private String type;
+    private String name;
+    private int number;
+    private double cost;
+
+    /**
+     * Constructor Product create object of product
+     * @param list List of products
+     */
+    public Product(ArrayList<String> list) {
```

Comment:
Do it in more natural way: obviously constructor must take 4 arguments - String type, String  name, int number, double price
This constructor should not relate to the way how you read data from console.